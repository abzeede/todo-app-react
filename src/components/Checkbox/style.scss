.checkbox {
  width: 20px;
  height: 20px;
  text-align: center;
  transition: 0.15s ease-in-out;
  border-radius: 15px;
  border: 1px solid #c4c7c7;
  color: #f3f3f3;
  background: #c4c7c7;
  position: initial;

  &::after {
    content: '\2713';
  }

  &--checked {
    background-color: #38b549;
    border-color: #38b549;
  }

  @media (hover: hover) {
    &:hover {
      background-color: #38b549;
      border-color: #38b549;
    }
  }

  @media (hover: hover) {
    &--checked:hover {
      border: 1px solid #c4c7c7;
      color: #f3f3f3;
      background: #c4c7c7;
    }
  }
}
