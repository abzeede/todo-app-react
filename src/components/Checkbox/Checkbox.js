import React from 'react'
import { bool, func } from 'prop-types'
import './style.scss'

const Checkbox = ({ checked, onClick }) => {
  return <span className={`checkbox ${checked && 'checkbox--checked'}`} onClick={onClick} />
}

Checkbox.propTypes = {
  checked: bool.isRequired,
  onClick: func.isRequired,
}

export default Checkbox
