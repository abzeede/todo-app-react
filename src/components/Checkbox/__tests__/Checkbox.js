import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Checkbox from '../Checkbox'

describe('Checkbox', () => {
  it('should render properly', () => {
    const checkedCheckbox = mount(<Checkbox onClick={() => {}} checked />)
    const unCheckedCheckbox = mount(<Checkbox onClick={() => {}} checked={false} />)

    expect(checkedCheckbox.find('.checkbox')).toHaveLength(1)
    expect(checkedCheckbox.find('.checkbox--checked')).toHaveLength(1)
    expect(toJson(checkedCheckbox)).toMatchSnapshot()

    expect(unCheckedCheckbox.find('.checkbox')).toHaveLength(1)
    expect(unCheckedCheckbox.find('.checkbox--checked')).toHaveLength(0)
    expect(toJson(unCheckedCheckbox)).toMatchSnapshot()
  })

  it(`should change call function onClick when it's clicked`, () => {
    const onClick = jest.fn(() => {})
    const checkedCheckbox = mount(<Checkbox onClick={onClick} checked />)

    checkedCheckbox.find('.checkbox').simulate('click')
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
