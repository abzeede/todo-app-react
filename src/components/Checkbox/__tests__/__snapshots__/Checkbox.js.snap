// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render properly 1`] = `
<Checkbox
  checked={true}
  onClick={[Function]}
>
  <span
    className="checkbox checkbox--checked"
    onClick={[Function]}
  />
</Checkbox>
`;

exports[`Checkbox should render properly 2`] = `
<Checkbox
  checked={false}
  onClick={[Function]}
>
  <span
    className="checkbox false"
    onClick={[Function]}
  />
</Checkbox>
`;
